notifications/src/main/java/dk/shape/games/notifications/presentation/SubjectNotificationsFragment.kt

```diff
@@ -0,0 +1,138 @@
+import dk.shape.games.notifications.R
+import dk.shape.games.notifications.actions.SubjectNotificationsAction
+import dk.shape.games.notifications.aliases.NotifificationsLoadedListener
+import dk.shape.games.notifications.databinding.FragmentSubjectNotificationsBinding
+import dk.shape.games.notifications.extensions.awareSet
+import dk.shape.games.notifications.extensions.launch
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationSheetViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationSwitcherViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationTypeCollectionViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationViewModel
+import dk.shape.games.notifications.usecases.SubjectNotificationUseCases
+import dk.shape.games.toolbox_library.configinjection.ConfigFragmentArgs
+import dk.shape.games.toolbox_library.configinjection.action
+import dk.shape.games.toolbox_library.configinjection.config
+import kotlinx.coroutines.Dispatchers
+
+class SubjectNotificationsFragment : BottomSheetDialogFragment() {
+
+    object Args : ConfigFragmentArgs<SubjectNotificationsAction, SubjectNotificationsConfig>()
+
+    private val config: SubjectNotificationsConfig by config()
+    private val action: SubjectNotificationsAction by action()
+
+    private val interactor: SubjectNotificationUseCases by lazy {
+        SubjectNotificationInteractor(
+            action = action,
+            provideDeviceId = config.provideDeviceId,
+            provideNotifications = config.provideNotifications,
+            notificationsDataSource = config.notificationsDataSource,
+            notificationsEventHandler = config.eventHandler
+        )
+    }
+
+    private val viewSwitcherViewModel: SubjectNotificationSwitcherViewModel by lazy {
+        SubjectNotificationSwitcherViewModel()
+    }
+
+    private val notificationViewModel: SubjectNotificationViewModel by lazy {
+        SubjectNotificationViewModel(
+            subjectId = action.subjectId,
+            subjectType = action.subjectType,
+            subjectName = action.subjectName,
+            onClosedPressed = {
+                config.eventHandler.onClosed(fragment = this, action = action)
+            },
+            onPreferencesSaved = { stateData, onSuccess, onFailure ->
+                launch(interactor, Dispatchers.IO) {
+                    whenResumed {
+                        saveNotificationPreferences(
+                            stateData = stateData,
+                            onSuccess = onSuccess,
+                            onFailure = { onFailure() }
+                        )
+                    }
+                }
+            }
+        )
+    }
+
+    private val notificationsSheetViewModel: SubjectNotificationSheetViewModel by lazy {
```

Comment:
```suggestion
    private val notificationsSheetViewModel = SubjectNotificationSheetViewModel( ...)
```
what is the benefit of using lazy in that case, since that view model is gonna be created as soon as onCreateView is called ?

Reply:
we need to wait to access action and config to avoid issues

Reply:
then I guess we have to do it that way unless we can defer view model creation to onViewCreated or onResume